daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/LayeredSequenceParser.scala

```diff
@@ -54,6 +55,8 @@ class LayeredSequenceParser(
     } catch {
       case le: LayerNotEnoughDataException =>
         PENotEnoughBits(state, le.schemaFileLocation, le.dataLocation, le.nBytesRequired * 8, MaybeULong.Nope)
+      case e: Exception =>
+        throw LayerExecutionException(s"Unexpected exception in layer transformer ${layerTransformer.layerName}: $e", e)
```

Comment:
I wonder if we should just throw an SDE? A layer is sort of an extension of a schema, so a layer doing something bad could maybe be consider similar to a bad schema and thus an SDE? Might simplify much of the logic since we can just rely on existing SDE handling to make the diagnostics available and users don't have to change any of their handling or need to catch a new exception type.

Reply:
I think the distinction is between "layer doing something bad" and the "user doing something bad". An SDE tells the 
user that bad input was provided and they can take action to correct it. In this case it is an actual bug in the layer which the developer should fix but the user wouldn't be able to. Wrapping this in an SDE might be confusing to the user.